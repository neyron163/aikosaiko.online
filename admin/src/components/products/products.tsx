import React from 'react';
// import {useQuery} from '@apollo/react-hooks';
import gql from 'graphql-tag';
import {graphql} from 'react-apollo';
import TextField from '@material-ui/core/TextField';
import {makeStyles} from '@material-ui/core/styles';

import s from './products.module.scss';

const useStyles = makeStyles(theme => ({
    container: {
        display: 'flex',
        flexWrap: 'wrap',
    },
    textField: {
        marginLeft: theme.spacing(1),
        marginRight: theme.spacing(1),
        width: 200,
    },
    dense: {
        marginTop: 19,
    },
    menu: {
        width: 200,
    },
}));

export const SET_PRODUCTS = gql`
    mutation($input: create) {
        createProduct(input: $input) {
            name
            img
            age
        }
    }
`;

/**
 * Products component
 */
export const Products = ({mutate}: any) => {
    const classes = useStyles();
    const [values, setValues] = React.useState({
        name: '',
        age: '',
        img: '',
    });

    /**
     * handleChange
     * @param name
     */
    const handleChange = (name: string) => (event: any) => {
        setValues({...values, [name]: event.target.value});
    };

    /**
     * sendData
     */
    function sendData(e: any) {
        e.preventDefault();
        const {name, age, img} = values;
        mutate({
            variables: {
                input: {
                    name,
                    age: Number(age),
                    img,
                },
            },
        });
        setValues({
            name: '',
            age: '',
            img: '',
        });
    }

    return (
        <div className={s.wrapper} onSubmit={e => sendData(e)}>
            <form>
                <TextField
                    id="standard-name"
                    label="Name"
                    className={classes.textField}
                    value={values.name}
                    onChange={handleChange('name')}
                    margin="normal"
                />
                <TextField
                    id="standard-name"
                    label="Age"
                    className={classes.textField}
                    value={values.age}
                    onChange={handleChange('age')}
                    margin="normal"
                />
                <TextField
                    id="standard-name"
                    label="Img"
                    className={classes.textField}
                    value={values.img}
                    onChange={handleChange('img')}
                    margin="normal"
                />
                <button type="submit">send</button>
            </form>
        </div>
    );
};
// @ts-ignore
export const ProductsQuery = graphql(SET_PRODUCTS)(Products);
