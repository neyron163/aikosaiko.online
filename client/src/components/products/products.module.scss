:local {
    .wrapper {
        display: flex;
        justify-content: space-between;
        padding: 50px 0;
        font-family: "Roboto", sans-serif;
        max-width: 1200px;
        margin-left: auto;
        margin-right: auto;
    }

    .leftBar {
        flex: 0 1 300px;
        height: 400px;
        margin: 0 10px;
        background: #e2e2e2;
        border-radius: 5px;
        box-shadow: 1px 0 10px 0 rgb(71, 61, 61);
    }

    .itemBox {
        display: flex;
        flex-wrap: wrap;
    }

    .item {
        max-width: 280px;
        box-sizing: border-box;
        margin: 0 20px 30px 0;
        background: #e2e2e2;
        border-radius: 5px;
        box-shadow: 1px 0 10px 0 rgb(71, 61, 61);
        background-color: #242525;
        cursor: pointer;
        transition: .3s transform;

        &:nth-child(4n) {
            margin-right: 0;
        }

        &:hover {
            transform: translateY(-10px);
        }

        img {
            min-width: 280px;
            min-height: 396px;
        }
    }

    .itemDescription {
        border-radius: 0 0 5px 5px;
        width: 100%;
        color: white;
        box-sizing: border-box;
        padding: 20px;
    }

    .itemNum {
        background: #4b8257;
        width: 30px;
        border-radius: 10px;
        padding: 0 3px;
        text-align: center;
        margin-right: 20px;
    }

    .top {
        display: flex;
        align-items: center;
        text-transform: uppercase;
    }

    .views {
        text-align: center;
        padding: 15px 0;
    }

    .itemFooter {
        display: flex;
        align-items: center;
        justify-content: space-between;
    }

    .svgBox {
        img {
            width: 30px;
        }
    }

    .rating {
        min-width: 20px;
        padding: 2px 10px;
        background-color: #c6a852;
        border-radius: 4px;
    }
}
