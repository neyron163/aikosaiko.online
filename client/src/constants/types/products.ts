export interface ProductsType {
    id: number;
    name: string;
    img: string;
    age: number;
    place: number;
    rating: string;
    votes: string;
    views: number;
    genre: Array<string>;
    type: string;
    year: number;
    status: string;
    studio: Array<string>;
    series: number;
}