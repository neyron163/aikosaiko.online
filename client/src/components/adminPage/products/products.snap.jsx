import React from 'react';
import renderer from 'react-test-renderer';

import {Products} from './products';

jest.mock('@material-ui/core/TextField', () => 'TextField');

describe('<Products />', () => {
    it('should render correctly', () => {
        const tree = renderer.create(<Products />);
        expect(tree.toJSON()).toMatchSnapshot();
    });
});
