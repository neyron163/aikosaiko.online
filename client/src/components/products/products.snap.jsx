import React from 'react';
import renderer from 'react-test-renderer';

import {Products} from './products';

let props;

beforeEach(() => {
    props = {
        data: {
            products: [
                {
                    id: 1,
                    name: 'name',
                    img: 'img',
                }
            ]
        },
        loading: false,
    }
})

describe('<Products />', () => {
    it('should render correctly', () => {
        const tree = renderer.create(<Products {...props} />);
        expect(tree.toJSON()).toMatchSnapshot();
    });

    it('should render correctly if loading true', () => {
        props.data.loading = true;
        const tree = renderer.create(<Products {...props} />);
        expect(tree.toJSON()).toMatchSnapshot();
    });
});
