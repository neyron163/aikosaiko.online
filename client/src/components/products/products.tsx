import React from 'react';
import gql from 'graphql-tag';
import {graphql} from 'react-apollo';
import {API_URL} from 'config';
import {BrowserRouter as Router, Link} from 'react-router-dom';
import {ProductsType} from 'constants/types/products';

import s from './products.module.scss';

export const GET_PRODUCTS = gql`
    query data {
        products {
            id
            name
            img
            views
            rating
            votes
            place
        }
    }
`;

interface PropsType {
    data: {
        products: ProductsType[];
        loading: boolean;
    };
}

/**
 * Products component
 */
export const Products: React.FC<PropsType> = ({data}) => {
    const {products, loading} = data!;
    if (loading) return null;

    return (
        <div className={s.wrapper}>
            <div className={s.itemBox}>
                {products.map(({id, name, img, views, rating, votes, place}: ProductsType) => (
                    <Link to={`/item/${id}`}>
                        <figure className={s.item} key={id}>
                            <img src={`${API_URL}${img}`} alt={name} />

                            <figcaption className={s.itemDescription}>
                                <div className={s.top}>
                                    <span className={s.itemNum}>{`# ${place}`}</span>
                                    <p className={s.itemName}>{name}</p>
                                </div>
                                <p className={s.views}>Просмотров: {views}</p>
                                <footer className={s.itemFooter}>
                                    <p className={s.rating}>{rating}</p>
                                    <p className={s.votes}>({votes} голосов)</p>
                                </footer>
                            </figcaption>
                        </figure>
                    </Link>
                ))}
            </div>
        </div>
    );
};

// export const ProductsQuery = graphql(GET_PRODUCTS)(Products);
export const ProductsQuery = Products;
