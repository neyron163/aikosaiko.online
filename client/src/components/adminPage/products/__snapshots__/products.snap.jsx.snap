// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Products /> should render correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <TextField
    className="makeStyles-textField-2"
    id="standard-name"
    label="Name"
    margin="normal"
    onChange={[Function]}
    value=""
  />
  <TextField
    className="makeStyles-textField-2"
    id="standard-name"
    label="Age"
    margin="normal"
    onChange={[Function]}
    value=""
  />
  <TextField
    className="makeStyles-textField-2"
    id="standard-name"
    label="Img"
    margin="normal"
    onChange={[Function]}
    value=""
  />
  <button
    type="submit"
  >
    send
  </button>
</form>
`;
